src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Tasks/DoNotUseWhenAllOrWaitAllWithSingleArgument.cs

```diff
@@ -112,7 +112,8 @@ private static bool IsSingleTaskArgument(IInvocationOperation invocation, INamed
                 return false;
             }
 
-            return namedTypeSymbol.Equals(taskType) || namedTypeSymbol.ConstructedFrom.Equals(genericTaskType);
+            return namedTypeSymbol.Equals(taskType, SymbolEqualityComparer.Default) ||
```

Comment:
I thought `symbol.Equals(otherSymbol)` implicitly used the `SymbolEqualityComparer.Default` if available?

Reply:
@ryzngard It does. But there is an analyzer (RS1024) to enforce using SymbolEqualityComparer explicitly to make the intent clear.

Currently the analyzer is disabled due to few bugs and will be re-enabled once they are fixed.